crabs/detection_tracking/evaluate_model.py

```diff
@@ -0,0 +1,161 @@
+import argparse
+import json
+import os
+
+import torch
+from detection_utils import (
+    create_dataloader,
+    get_test_transform,
+    myFasterRCNNDataset,
+)
+from evaluate import test_detection
+
+# select device (whether GPU or CPU)
+device = (
+    torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+)
+
+
+class Detector_Test:
+    """
+    A class for testing object detection models using pre-trained classification.
+
+    Args:
+        args (argparse.Namespace): Command-line arguments containing
+        configuration settings.
+
+    Attributes:
+        args (argparse.Namespace): The command-line arguments provided.
+        main_dir (str): The main directory path.
+        ious_threshold (float): The ious threshold for detection bounding boxes.
+        trained_model: The pre-trained subject classification model.
+        test_data (str): The path to the directory containing test images.
+        test_label (str): The path to the test annotation JSON file.
+        test_dataset: An instance of myFasterRCNNDataset for test data.
+        test_dataloader: The DataLoader for the test dataset.
+
+    Methods:
+        _load_pretrain_model(self) -> None:
+            Load the pre-trained subject classification model.
+
+        _load_dataset(self) -> None:
+            Load images and annotation file for testing.
+
+        test_model(self) -> None:
+            Test the pre-trained model on the test dataset.
+
+    """
+
+    def __init__(self, args: argparse.Namespace) -> None:
+        self.args = args
+        self.main_dir = args.main_dir
+        self.ious_threshold = args.ious_threshold
+
+    def _load_pretrain_model(self) -> None:
+        """
+        Load the pre-trained subject classification model.
+        """
+        # Load the pre-trained subject predictor
+        # TODO: deal with different model
+        self.trained_model = torch.load(
+            self.args.model_dir, map_location=torch.device("cpu")
+        )
+
+    def _load_dataset(self) -> None:
```

Comment:
can we reuse the equivalent code from the detector utils? Maybe we can have a parent class that the evaluation class and the detector class inherit from? Right now it feels a bit duplicated